### PR TITLE
Updates to packages with vulnerabilities and upgrade to latest Plaid package

### PR DESCRIPTION
**Pull request by kylehild**
There were a handful of vulnerabilities when running `npm install` so I fixed those. Fixing those also resulted in updating Plaid to the newest version which included breaking changes to creating the client as well as all of the requests. Those have been updated in this pull request as well.

### CODE REVIEW

File: actual.js [hunk 1]
```diff
@@ -30,7 +30,12 @@ async function initialize(config) {
             config.set("budget_id", id)
         }
 
-        await actual.downloadBudget(id);
+        if (appConfig.ACTUAL_SERVER_IS_ENCRYPTED) {
+            await actual.downloadBudget(id, { password: appConfig.ACTUAL_SERVER_PASSWORD });
```
Comment (youngcw) [inline actual.js:34]: Is this assuming that the file password is the same as the server password?  Shouldn't there be a new variable for the encryption password?

**Comment by kylehild** [comment]
Yeah, I can make that a new environment variable. I'll update the base address for the fastify server to be a variable as well. On May 17, 2024 11:06 AM, youngcw ***@***.***> wrote:
@youngcw commented on this pull request.



In actual.js:
> @@ -30,7 +30,12 @@ async function initialize(config) {
             config.set("budget_id", id)
         }
 
-        await actual.downloadBudget(id);
+        if (appConfig.ACTUAL_SERVER_IS_ENCRYPTED) {
+            await actual.downloadBudget(id, { password: appConfig.ACTUAL_SERVER_PASSWORD });

Is this assuming that the file password is the same as the server password?  Shouldn't there be a new variable for the encryption password?

File: actual.js (hunk 1 above)
Comment (kylehild) [reply actual.js]: I've updated this.

File: .env.sample [hunk 2]
```diff
@@ -17,9 +17,11 @@ PLAID_LANGUAGE="en"
 
 ACTUAL_SERVER_URL="server url"
 ACTUAL_SERVER_PASSWORD="server password"
+ACTUAL_SERVER_ENCRYPTION_PASSWORD="server password"
```
Comment (youngcw) [inline .env.sample:20]: Wont this example value cause the script to always think that budget file is encrypted?  I think it should either be blank by default, or there should still be a boolean for encrypted.

**Pull request merged**